2_standard_libraries/itertools_examples.py

```diff
@@ -0,0 +1,36 @@
+from itertools import groupby, takewhile, dropwhile
+
+
+def do_group_by():
+    #  groupby returns consecutive keys and groups / lot's of possible uses here but too advanced now
+    my_list = [('crank', 'shimano'), ('frame', 'bmc'), ('frame', 'giant'),
+               ('crank', 'sram')]
+    for brands, components in groupby(my_list, lambda x: x[
+        0]):  # lambda tells groupby to use first item as grouping key
+        for item in components:
+            print(f'We have a {brands} by {item[1]}')
+
+
+def do_takewhile():  # ???? I don't understand this, it's not iterating and just stops at the first odd number
```

Comment:
```
Make an iterator that returns elements from the iterable as long as the predicate is true.
```
in your case, the predicate is true only for the first element. the next one ("3") returns `false`, hence we stop iterating.